exercises/allergies/allergies.dpr

```diff
@@ -1,4 +1,4 @@
-program AllergyTests;
+program allergies;
```

Comment:
@kotp my 2nd and 3rd commits take care of the issue.  I tried squashing the 2nd and 3rd together but that kills the file rename and stuck with the original lowercase filename.  Hopefully when this PR is squashed and merged the rename takes.